### DISCUSSION TITLE
Add more fetch options to HttpRequestOptions

### DISCUSSION THREAD

**Discussion by lwansbrough** [panva/oauth4webapi#79] [Ideas]
Based on the fetch API, I would like to be able to specify a couple more options to pass to the fetch call during a `discoveryRequest` call. In particular, I'd like to be able to specify the `cache` and `mode`. Perhaps a way to specify all fetch options in an object, which would then be spread during the fetch invocation.

**Comment by panva** [comment]
Can you elaborate on why this is needed?

**Comment by lwansbrough** [comment]
@panva Certainly. We expect extremely high resiliency for our auth process -- across tens of millions of MAU, many users end up having clock skew. In order to determine exactly how much clock skew they have, we can use the server's time. You can create a time API for this, but that's an extra request/additional RTT to the server. Alternatively, you can use the `Date` header provided by many servers (or Cloudflare in our case) which is always up to date. The problem is, `fetch` will cache this request by default. So subsequent calls to the discovery endpoint may be served from disk, which means the Date header would be out of date.

I'm starting to lean towards just doing the API, as handling caching is always a pain. But there may be other legitimate reasons to allow specifying some of those options.

**Reply by panva** [reply]
Or you can just do the fetch call to the well-known endpoint with any options you want. Then proceed to pass the Response object to the function that usually follows. 

The module's api is split like so exactly to allow such customizations.

**Reply by lwansbrough** [reply]
I see, thanks.

**Comment by panva** [comment]
I'm looking for feedback for #94 which would enable you to pass whatever you want to fetch.
